### PR TITLE
ui/ performance optimization

### PR DESCRIPTION
**Pull request by leon-up9**

### CODE REVIEW

File: ui/src/components/PathList/PathList.tsx [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
 
-export const PathList: React.FC<Props> = ({ changeList }) => {
+const PathList: React.FC<Props> = ({ changeList }) => {
     const [type, setType] = useState('')
     const [path, setPath] = useState('')
-
-    const onPathFilterChange = (setFunc: Function) => (event: SelectChangeEvent<string> | React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => setFunc(event.target.value)
+    const debouncedSearchTerm = useDebounce(path, 200);
+    const onChangeTypeChange = (event) => { setType(event.target.value); }
+    const onPathchange = (event) => { setPath(event.target.value); }
```
Comment (lirazyehezkel) [inline ui/src/components/PathList/PathList.tsx]: onPathChange

File: ui/src/components/PathList/PathList.tsx [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
 
     const filteredChanges = useMemo(() => {
```
Comment (lirazyehezkel) [inline ui/src/components/PathList/PathList.tsx]: Try to find a better name - maybe filteredListItems ?

File: ui/src/components/PathList/PathList.tsx [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
+import useDebounce from '../../hooks/useDebounce';
 
 export interface Props {
     changeList: DataItem[]
 }
 
-export const PathList: React.FC<Props> = ({ changeList }) => {
+const PathList: React.FC<Props> = ({ changeList }) => {
     const [type, setType] = useState('')
     const [path, setPath] = useState('')
```
Comment (lirazyehezkel) [inline ui/src/components/PathList/PathList.tsx]: is it the path filter?
if it is, let's rename to pathFilter

File: ui/src/components/PathList/PathList.tsx [hunk 4]
```diff
@@ -1,27 +1,28 @@
...
 import { ChangeTypeEnum } from '../../consts';
+import useDebounce from '../../hooks/useDebounce';
 
 export interface Props {
     changeList: DataItem[]
 }
 
-export const PathList: React.FC<Props> = ({ changeList }) => {
+const PathList: React.FC<Props> = ({ changeList }) => {
     const [type, setType] = useState('')
```
Comment (lirazyehezkel) [inline ui/src/components/PathList/PathList.tsx]: rename to typeFilter

File: ui/src/components/PathList/PathList.tsx [hunk 5]
```diff
@@ -31,15 +32,15 @@ export const PathList: React.FC<Props> = ({ changeList }) => {
...
+                        <TextField id="outlined-basic" label="Path" variant="outlined" size="small" value={path} onChange={onPathchange} />
                     </FormControl>
                     <div className='seperatorLine'></div>
                     <FormControl size='small' sx={{ minWidth: 150 }} >
                         <InputLabel>Change Type</InputLabel>
                         <Select
                             label="Change Type"
                             value={type}
-                            onChange={onPathFilterChange(setType)}
+                            onChange={onChangeTypeChange}
```
Comment (lirazyehezkel) [inline ui/src/components/PathList/PathList.tsx]: onTypeChange might be a better name

File: ui/src/components/PathList/PathList.tsx [hunk 6]
```diff
@@ -53,11 +54,13 @@ export const PathList: React.FC<Props> = ({ changeList }) => {
                     </FormControl>
                 </div>
                 <div className='changeLogList'>
-                    {filteredChanges?.map((change: DataItem, index: number) => <div key={"changeLogItem" + index} className='changeLogItem'>
-                        <PathListItem changeLogItem={change} showChangeType={type} /></div>)
-                    }
+                    {filteredChanges?.map((change: DataItem, index: number) =>
+                        <PathListItem key={index} changeLogItem={change} showChangeType={type} />
```
Comment (lirazyehezkel) [inline ui/src/components/PathList/PathList.tsx]: it is not best practice to use index for react element key..
do we have maybe a key or some identifier for each dataItem that we can use instead?

File: ui/src/components/PathList/PathListItem.tsx [hunk 7]
```diff
@@ -1,81 +1,38 @@
...
-import { infoClass, ChangeTypeEnum } from "../../consts";
-import { SyntaxHighlighter } from "../SyntaxHighlighter";
+import { PathDisplay } from "./PathDisplay";
+import React from "react";
+import { ChangeTypeEnum } from "../../consts";
 
 export interface PathListItemProps {
     changeLogItem: DataItem
     showChangeType?: string
+    style?: any
```
Comment (lirazyehezkel) [inline ui/src/components/PathList/PathListItem.tsx]: why do we get style prop but we never use it?

File: ui/src/components/PathList/PathListItem.tsx [hunk 8]
```diff
@@ -1,81 +1,38 @@
...
-            case ChangeTypeEnum.Created:
-                return ChangeTypeEnum.Created
-            case ChangeTypeEnum.Updated:
-                return ChangeTypeEnum.Deleted
-            case ChangeTypeEnum.Deleted:
-                return ChangeTypeEnum.Deleted
-            default:
-                return infoClass
-        }
+    const onClick = () => {
```
Comment (lirazyehezkel) [inline ui/src/components/PathList/PathListItem.tsx]: maybe onAccordionClick would be a better name

File: ui/src/hooks/useDebounce.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from "react";
+
+const useDebounce = (value, delay) => {
```
Comment (lirazyehezkel) [inline ui/src/hooks/useDebounce.ts:3]: why can't we use lodash debounce or something else that already exists?

File: ui/src/components/PathList/PathDisplay.tsx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+import { Path } from "../../interfaces"
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import SyntaxHighlighter from "../SyntaxHighlighter"
+import { ChangeTypeEnum, infoClass } from "../../consts";
+import { useState } from "react";
+
+
+export interface PathDisplayProps {
+    path: Path
+    style?: any
```
Comment (lirazyehezkel) [inline ui/src/components/PathList/PathDisplay.tsx]: why do we get style prop but we never use it?

**Review by lirazyehezkel — COMMENTED** [review]
added some comments

File: ui/src/components/PathList/PathList.tsx (hunk 6 above)
Comment (leon-up9) [reply ui/src/components/PathList/PathList.tsx]: i know...but we dont have identifier.
should i inject it once the data injected?

File: ui/src/hooks/useDebounce.ts (hunk 9 above)
Comment (leon-up9) [reply ui/src/hooks/useDebounce.ts:3]: its the solution i found to seperate the state update and the filter operation.
I as see it the trade off will be to hold another variable and make the debounce on him and useEffect to listen to that change to make the filter operation.
I come from the assumption, the state must be updated immediately

**Review by lirazyehezkel — APPROVED** [review]

**Pull request merged**